### ISSUE TITLE
App crashes on Windows, do I need a valid connection string to launch it?

### ISSUE THREAD

**Issue by alsorokin** [Azure/azure-mobile-engagement-cordova#36]
Just run the following commands to repro it:

```
cordova create foo
cd foo
cordova platform add windows
cordova plugin add cordova-plugin-ms-azure-mobile-engagement --variable AZME_WINDOWS_CONNECTION_STRING=test
cordova run
```

**Comment by ogoguel** [comment]
No, a valid string is not required to launch your app (but obviously, the connection to the azme backend will fail!)

However, you need to execute "cordova build" before running your app, as there's some processing in the "after_build.js" hook . without that processing, the EngagementProxy.js is invalid, hence the exeception.

Normally, a "cordova run" command does a "build", but i dunno why the hook is not executed in that case.

**Comment by alsorokin** [comment]
Running "cordova build" before "cordova run" doesn't seem to have fixed the issue.

I also see the following error in the "Applications and Services Logs/Microsoft/Windows/AppHpst/AppTracing" event log:
SCRIPT5009 : '$AZME_ENABLE_PLUGIN_LOG' is undefined

**Comment by ogoguel** [comment]
You're right, the postprocessing script is not being called at the right : it should be "after the prepare" step, and not "after the build step"

I'm preparing a fix for that : thanks for the pointer

**Comment by ogoguel** [comment]
Please check PR to confirm that it solves your issue : https://github.com/Azure/azure-mobile-engagement-cordova/pull/33

**Comment by alsorokin** [comment]
No luck :( The crash is still happening, the error message in the event log is the same.
I tried adding a plugin, then `cordova prepare`, `cordova build`, `cordova run`

**Comment by alsorokin** [comment]
Here's the full log, if it helps:
```
C:\Projects\Cordova>cordova create foo
Warning: using prerelease version 6.4.0-dev (cordova-lib@6.4.0-dev)
Using detached cordova-create
Creating a new cordova project.

C:\Projects\Cordova>cd foo

C:\Projects\Cordova\foo>cordova platform add windows
Warning: using prerelease version 6.4.0-dev (cordova-lib@6.4.0-dev)
Adding windows project...
Creating Cordova Windows Project:
        Path: platforms\windows
        Namespace: io.cordova.hellocordova
        Name: HelloCordova
Windows project created with cordova-windows@4.4.2
Discovered plugin "cordova-plugin-whitelist" in config.xml. Adding it to the project
Fetching plugin "cordova-plugin-whitelist@1" via npm
Installing "cordova-plugin-whitelist" for windows

C:\Projects\Cordova\foo>cordova plugin add https://github.com/ogoguel/azure-mobile-engagement-cordova --variable AZME_WINDOWS_CONNECTION_STRING=test
Warning: using prerelease version 6.4.0-dev (cordova-lib@6.4.0-dev)
Fetching plugin "https://github.com/ogoguel/azure-mobile-engagement-cordova" via git clone
Repository "https://github.com/ogoguel/azure-mobile-engagement-cordova" checked out to git ref "master".
Installing "cordova-plugin-ms-azure-mobile-engagement" for windows

C:\Projects\Cordova\foo>cordova prepare
Warning: using prerelease version 6.4.0-dev (cordova-lib@6.4.0-dev)

C:\Projects\Cordova\foo>cordova build
Warning: using prerelease version 6.4.0-dev (cordova-lib@6.4.0-dev)
Building project: C:\Projects\Cordova\foo\platforms\windows\CordovaApp.Windows.jsproj
        Configuration : debug
        Platform      : anycpu
  CordovaApp.Windows -> C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Windows_1.0.0.0_anycpu_debug_Test\CordovaApp.Windows_1.0.0.0
  _anycpu_debug.appx
  CordovaApp.Windows -> C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Windows_1.0.0.0_anycpu_debug.appxupload
  Your package has been successfully created.
Building project: C:\Projects\Cordova\foo\platforms\windows\CordovaApp.Phone.jsproj
        Configuration : debug
        Platform      : anycpu
  CordovaApp.Phone -> C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Phone_1.0.0.0_anycpu_debug_Test\CordovaApp.Phone_1.0.0.0_anycp
  u_debug.appx
  CordovaApp.Phone -> C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Phone_1.0.0.0_anycpu_debug.appxupload
  Your package has been successfully created.

C:\Projects\Cordova\foo>cordova run
Warning: using prerelease version 6.4.0-dev (cordova-lib@6.4.0-dev)
Building project: C:\Projects\Cordova\foo\platforms\windows\CordovaApp.Windows.jsproj
        Configuration : debug
        Platform      : anycpu
  CordovaApp.Windows -> C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Windows_1.0.0.0_anycpu_debug_Test\CordovaApp.Windows_1.0.0.0
  _anycpu_debug.appx
  CordovaApp.Windows -> C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Windows_1.0.0.0_anycpu_debug.appxupload
  Your package has been successfully created.
Building project: C:\Projects\Cordova\foo\platforms\windows\CordovaApp.Phone.jsproj
        Configuration : debug
        Platform      : anycpu
  CordovaApp.Phone -> C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Phone_1.0.0.0_anycpu_debug_Test\CordovaApp.Phone_1.0.0.0_anycp
  u_debug.appx
  CordovaApp.Phone -> C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Phone_1.0.0.0_anycpu_debug.appxupload
  Your package has been successfully created.
Deploying windows package to device:
C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Windows_1.0.0.0_anycpu_debug_Test\CordovaApp.Windows_1.0.0.0_anycpu_debug.appx
Attempting to uninstall previous application version...
Attempting to install application...
Found package: C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Windows_1.0.0.0_anycpu_debug_Test\CordovaApp.Windows_1.0.0.0_anycpu_debug.appx

Installing app...
Found dependency package(s):
C:\Projects\Cordova\foo\platforms\windows\AppPackages\CordovaApp.Windows_1.0.0.0_anycpu_debug_Test\Dependencies\Microsoft.WinJS.2.0.appx

Success: Your app was successfully installed.
Starting application...
ActivateApplication:  io.cordova.hellocordova_h35559jr9hy9m!io.cordova.hellocordova
```

**Comment by ogoguel** [comment]
OK, this time I got it : thanks for having sent those details logs, appreciated!

It crashed because the variables with a default value (`AZME_ENABLE_NATIVE_LOG` and `AZME_ENABLE_PLUGIN_LOG`) were not properly replaced in the `.js` file... if not defined during the plugin installation. 

FIxed in the PR #33

Note that your application is still going to crash at first run, but this time, it will be due to an exception while parsing the connection string.  If you want to bypass it, you should use  `Endpoint=0;AppId=0;SdkKey=0` instead of `test`as the connection string.

Hope everything will be fine this time!

**Comment by alsorokin** [comment]
It works, thanks @ogoguel 
Good stuff!
